schema-engine/cli/tests/cli_tests.rs

```diff
@@ -16,69 +16,96 @@ fn schema_engine_bin_path() -> &'static str {
     env!("CARGO_BIN_EXE_schema-engine")
 }
 
-fn run(args: &[&str]) -> Output {
+trait CommandExt {
+    fn datasource_url(&mut self, url: impl Into<String>) -> &mut Self;
+}
+
+impl CommandExt for Command {
+    fn datasource_url(&mut self, url: impl Into<String>) -> &mut Self {
+        self.arg("--datasource")
+            .arg(serde_json::to_string(&DatasourceUrls::from_url(url)).unwrap())
+    }
+}
+
+fn run_with_datasource_url(url: impl Into<String>, args: &[&str]) -> Output {
     Command::new(schema_engine_bin_path())
+        .datasource_url(url)
         .arg("cli")
         .args(args)
         .env("RUST_LOG", "INFO")
         .output()
         .unwrap()
 }
 
+struct DropChild(Child);
+
+impl DropChild {
+    fn get_mut(&mut self) -> &mut Child {
+        &mut self.0
+    }
+}
+
+impl Drop for DropChild {
+    fn drop(&mut self) {
+        _ = self.0.kill();
+    }
+}
+
 fn with_child_process<F>(mut command: Command, f: F)
 where
     F: FnOnce(&mut Child),
 {
-    let mut child = command
+    let child = command
         .stdin(std::process::Stdio::piped())
         .stdout(std::process::Stdio::piped())
         .stderr(std::process::Stdio::null())
         .spawn()
         .unwrap();
 
-    let res = panic::catch_unwind(AssertUnwindSafe(|| {
-        f(&mut child);
-    }));
-
-    child.kill().unwrap();
-    match res {
-        Ok(_) => (),
-        Err(panic_payload) => {
-            panic!(
-                "Error: '{}'",
-                panic_utils::downcast_box_to_string(panic_payload).unwrap_or_default()
-            );
-        }
-    }
+    let mut child = DropChild(child);
+
+    f(child.get_mut());
```

Comment:
Did this refactoring when debugging a failing test because I was confused by an empty panic message and thought `panic_utils::downcast_box_to_string(panic_payload)` didn't work.

Turned out that the empty panic message was correct because `expect-test` does some weird stuff with its own custom panic implementation and prints the failure messages to `stdout` (not even `stderr`) using `println!`.

However, I left this change because this implementation is better and more idiomatic, and the previous implementation approached the problem backwards.

Reply:
>Turned out that the empty panic message was correct because expect-test does some weird stuff with its own custom panic implementation and prints the failure messages to stdout (not even stderr) using println!.

😩